basicsr/models/base_model.py

```diff
@@ -43,6 +43,11 @@ def validation(self, dataloader, current_iter, tb_logger, save_img=False):
             tb_logger (tensorboard logger): Tensorboard logger.
             save_img (bool): Whether to save images. Default: False.
         """
+        # 20210523 [Lotayou]: Critical patch: HiFaceGAN requires train() mode even for validation
+        # For more info, see https://github.com/Lotayou/Face-Renovation/issues/31
+        if self.opt['network_g']['type'] in ('HiFaceGAN', 'SPADEGenerator'):
```

Comment:
I recommend to implement this in the model file instead of hard-coding in the basemodel class.

may be like this
```python
def nondist_validation(self, dataloader, current_iter, tb_logger, save_img):
        self.is_train = False
        super(xxx, self).nondist_validation(dataloader, current_iter, tb_logger, save_img)
        self.is_train = True
```